### ISSUE TITLE
need support to add categorical variables for stratifications in lineup for filtering

### ISSUE THREAD

**Issue by ngehlenborg** [Caleydo/caleydo#1448]
I would like to load a categorical variable associated with genes (which in the case of mutations/cnv correspond to stratifications) and I want to filter on that categorical variable.

The variable in question tells me whether a gene is a tumor suppressor gene or an oncogene. Not all gene symbols might match.

Data from http://www.sciencemag.org/content/339/6127/1546.

```
Gene Symbol,Classification
ABL1,Oncogene
ACVR1B,TSG
AKT1,Oncogene
ALK,Oncogene
APC,TSG
AR,Oncogene
ARID1A,TSG
ARID1B,TSG
ARID2,TSG
ASXL1,TSG
ATM,TSG
ATRX,TSG
AXIN1,TSG
B2M,TSG
BAP1,TSG
BCL2,Oncogene
BCOR,TSG
BRAF,Oncogene
BRCA1,TSG
BRCA2,TSG
CARD11,Oncogene
CASP8,TSG
CBL,Oncogene
CDC73,TSG
CDH1,TSG
CDKN2A,TSG
CEBPA,TSG
CIC,TSG
CREBBP,TSG
CRLF2,Oncogene
CSF1R,Oncogene
CTNNB1,Oncogene
CYLD,TSG
DAXX,TSG
DNMT1,Oncogene
DNMT3A,Oncogene
EGFR,Oncogene
EP300,TSG
ERBB2,Oncogene
EZH2,Oncogene
FAM123B,TSG
FBXW7,TSG
FGFR2,Oncogene
FGFR3,Oncogene
FLT3,Oncogene
FOXL2,Oncogene
FUBP1,TSG
GATA1,TSG
GATA2,Oncogene
GATA3,TSG
GNA11,Oncogene
GNAQ,Oncogene
GNAS,Oncogene
H3F3A,Oncogene
HIST1H3B,Oncogene
HNF1A,TSG
HRAS,Oncogene
IDH1,Oncogene
IDH2,Oncogene
JAK1,Oncogene
JAK2,Oncogene
JAK3,Oncogene
KDM5C,TSG
KDM6A,TSG
KIT,Oncogene
KLF4,Oncogene
KRAS,Oncogene
MAP2K1,Oncogene
MAP3K1,TSG
MED12,Oncogene
MEN1,TSG
MET,Oncogene
MLH1,TSG
MLL2,TSG
MLL3,TSG
MPL,Oncogene
MSH2,TSG
MSH6,TSG
MYD88,Oncogene
NCOR1,TSG
NF1,TSG
NF2,TSG
NFE2L2,Oncogene
NOTCH1,TSG
NOTCH2,TSG
NPM1,TSG
NRAS,Oncogene
PAX5,TSG
PBRM1,TSG
PDGFRA,Oncogene
PHF6,TSG
PIK3CA,Oncogene
PIK3R1,TSG
PPP2R1A,Oncogene
PRDM1,TSG
PTCH1,TSG
PTEN,TSG
PTPN11,Oncogene
RB1,TSG
RET,Oncogene
RNF43,TSG
RUNX1,TSG
SETD2,TSG
SETBP1,Oncogene
SF3B1,Oncogene
SMAD2,TSG
SMAD4,TSG
SMARCA4,TSG
SMARCB1,TSG
SMO,Oncogene
SOCS1,TSG
SOX9,TSG
SPOP,Oncogene
SRSF2,Oncogene
STAG2,TSG
STK11,TSG
TET2,TSG
TNFAIP3,TSG
TRAF7,TSG
TP53,TSG
TSC1,TSG
TSHR,Oncogene
U2AF1,Oncogene
VHL,TSG
WT1,TSG
```

**Comment by sgratzl** [comment]
In this first version, groupings for categorical data, will be automatically added as another metric column. 

If you need more, please tell me

So you first have to import this grouping to the data domain, e.g. Mutations and then if you select the Mutations database it should appear as another metric column

**Comment by ngehlenborg** [comment]
This is working quite well, but I have two observations:
- [x] The new column was added to the Copy Number Metrics (I did load it for the copy number data domain, but it should show up as a separate column and not be mixed with copy number levels).
- [x] The above example contains entries for only a small subset of the genes, i.e., a lot genes will be assigned "NA", while some will have "Oncogene" or "TSG". However, the filter for the column only contains the latter two categories but not "NA", but I would like to remove exactly those genes using the filter. We need to add the "NA" category, maybe using "(none)" as the label.
